airbyte-integrations/connectors/destination-pinecone/destination_pinecone/document_processor.py

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import List, Mapping, Optional, Tuple, Union
+
+import dpath.util
+from airbyte_cdk.models import AirbyteRecordMessage, ConfiguredAirbyteCatalog, ConfiguredAirbyteStream
+from airbyte_cdk.models.airbyte_protocol import AirbyteStream, DestinationSyncMode
+from destination_pinecone.config import ProcessingConfigModel
+from dpath.exceptions import PathNotFound
+from langchain.document_loaders.base import Document
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.utils import stringify_dict
+
+METADATA_STREAM_FIELD = "_airbyte_stream"
+METADATA_RECORD_ID_FIELD = "_record_id"
+
+
+class DocumentProcessor:
+    streams: Mapping[str, ConfiguredAirbyteStream]
+
+    def __init__(self, config: ProcessingConfigModel, catalog: ConfiguredAirbyteCatalog, max_metadata_size: Optional[int] = None):
+        self.streams = {self._stream_identifier(stream.stream): stream for stream in catalog.streams}
+        self.max_metadata_size = max_metadata_size
+
+        self.splitter = RecursiveCharacterTextSplitter.from_tiktoken_encoder(
```

Comment:
This is hardcoded to tiktoken encoder, but that's more confusing than helpful when the cohere embeddings are used as they don't use the same token representation, right? Should we just go by characters here instead? That would be faster and has also a lower chance for confusion.

Reply:
I think we can leave it as is for now, I think the differences are negligeable, but I'll look into this further.